pontis-package/pontis/core/base_client.py

```diff
@@ -40,8 +40,8 @@ def __init__(
         self.signer = Signer(self.account_private_key)
 
         self.client = Client(self.network, n_retries=n_retries)
+        self.latest_nonce = None
```

Comment:
what is the difference between ```self.latest_nonce``` and ```self.nonce``` in this context?

Reply:
If I understand it correctly, ```self.nonce``` is for the particular tx being sent but ```self.latest_nonce``` is just the last nonce of the last tx sent from the contract had confirmed (and it is updated or set to pending depending on the outcome of whatever had been broadcasted to the meempool + our pending flag).

```self.nonce``` seems to used more for error handling bc the network won't update fast enough for us to query for the next nonce. Line 88 made it make sense for me.

 Is it possible we can just use ```self.latest``` and switch between updating it locally (iterate + 1) and pulling from the network (which ```self.latest``` does now. Pretty low impact change, just thinking out loud, good anyways.

Reply:
`self.nonce` is what we think the nonce is of the next transaction to be submitted.
`self.latest_nonce` is the nonce of the latest confirmed transaction (should maybe be called `latest_confirmed_nonce`). We need the nonce in the latest confirmed batch, because that is what we need to use for the `estimate_fee` call, which runs the tx on the network state after the last confirmed block. 

For actually submitting the tx, we want to use `nonce`, which is the pending nonce (with our local caching). 

The issue is described on line 54 - let me know how you feel about renaming to `self.latest_confirmed_nonce` or updating the comment to clarify.

Reply:
Noted. I think the confusion was about ```self.nonce``` which might more appropriately be named ```self.next_nonce```. When I saw ```self.nonce``` attached as an instance variable of the client (and not an instance of a tx) my bias was to think that references the last confirmed nonce, not the upcoming one. 

In any case the label ```upcoming_nonce``` or ```next_nonce``` for ```self.nonce``` irons out that ambiguity. Alternatively I am not opposed to having one nonce that updates situationally, and the error handling where we set ```self.nonce = None``` in (and to indicate) failed transactions, to refer to a nonce which is an instance variable of the particular tx, not the client. 

See line 59 for this second raised issue on disaggregating account nonce, and TODO getters that would return the nonce of a particular tx, with error handling for failed txs.